kotlinpoet/src/main/java/com/squareup/kotlinpoet/CodeWriter.kt

```diff
@@ -628,18 +628,18 @@ internal class CodeWriter constructor(
    * there are no other opposing modifiers (like [KModifier.PROTECTED] etc.) supplied by the
    * consumer.
    */
-  private fun shouldEmitPublicModifier(
+  private fun shouldEmitImplicitPublicModifier(
```

Comment:
Not sure I agree with the name change, it looks like the KDoc is incorrect - this method does not return true when `public` is in the set of implicit modifiers. My suggestion would be to revert the name change, and to delete the KDoc entirely - it's not needed for a private method, and the code is readable enough to not warrant extra documentation.

Reply:
Just to make sure we're on the same page - the method definitely can return true if public is in the set of implicit modifiers. In fact after my change it will only return true if public is in implicit modifiers (and no other opposing modifiers...).

With that in mind, if you think the original name is clearer and it's better to delete the kdoc, that's fine with me.

It's a bit confusing because implicitModifiers is mostly used to skip explicit modifiers, except in the case of public where it's used to add public even if it's not explicit.

Reply:
Yeah, you're right - just re-read the method. It is indeed confusing.